datadog/resource_datadog_monitor.go

```diff
@@ -132,6 +132,7 @@ func resourceDatadogMonitor() *schema.Resource {
 			"no_data_timeframe": {
 				Type:     schema.TypeInt,
 				Optional: true,
+				Default:  10,
```

Comment:
We should use `DefaultFunc` and only set this when `notify_no_data` is `true`, I think.

Also, we could actually try parsing out the evaluation timeframe from the query and set it to the recommended values from the docs [1] - let's fall back to 10 minutes if fail to parse out the evaluation timeframe.

[1] https://docs.datadoghq.com/api/?lang=python#monitors

Reply:
As discussed offline a bit, lets go with a generically default 10 minutes and just add a note that it should be set to 2x the evaluation window. We can tackle something more complex if its needed later 😄

Reply:
Actually, the docs have a pretty clear note that the timeframe, if specified, should be at least 2x the eval window. Do you think this is good to be merged?

Reply:
Yeah, I'm ok with merging this :+1: